src/FixedDataTableBufferedRows.js

```diff
@@ -78,27 +81,32 @@ class FixedDataTableBufferedRows extends React.Component {
     var rowClassNameGetter = props.rowClassNameGetter || emptyFunction;
     var rowsToRender = this.props.rowsToRender || [];
 
-    this._staticRowArray.length = rowsToRender.length;
+    if (props.isScrolling) {
+      // we are scrolling, so don't display any rows which lie outside the viewport
+      this._staticRowArray.forEach((row, i) => {
+        const rowOutsideViewport = !this.isRowInsideViewport(row.props.index);
+        if (rowOutsideViewport) {
+          this._staticRowArray[i] = React.cloneElement(this._staticRowArray[i], {
+            visible: false,
+          });
+        }
+      });
+    } else {
+      this._staticRowArray.length = rowsToRender.length;
+    }
+
     var baseOffsetTop = props.offsetTop - props.scrollTop;
 
-    for (let i = 0; i < rowsToRender.length; ++i) {
+    for (let i = 0; i < rowsToRender.length; i++) {
```

Comment:
From the previous review:
> Since we use a forEach, only defined values (rows) are iterated over.

forEach also includes undefined values.  Try `[0, '0', undefined, null, 1, '1'].forEach(console.log)`

Thanks for adding back the undefined handling

Reply:
Aah my bad, I totally wrote that wrong.
What I meant was that only indices assigned are iterated.
```
a = [];
a[0] = 0;
a[1] = 1;
a[3] = 3;
a.forEach(console.log); // won't iterate for a[2].
```

Reply:
ah okay that makes sense.  As long as everything still works if we have
a[2] = undefined;